I managed to install FreeBSD 15 in Virtualbox. First time I ever try to do this.

I got similar results as @cqundefine, but I documented my experience as a beginner with FreeBSD. @cqundefine would you be able to comment if you are seeing similar issues or if my config must be all messed up.

Nice to see that it is using Clang 19 as the default compiler.

```
free@FreeBSD:~/projects/ladybird $ cc -v
FreeBSD clang version 19.1.7 (https://github.com/llvm/llvm-project.git llvmorg-19.1.7-0-gcd708029e0b2)
Target: x86_64-unknown-freebsd15.0
Thread model: posix
InstalledDir: /usr/bin
free@FreeBSD:~/projects/ladybird $ c++ -v
FreeBSD clang version 19.1.7 (https://github.com/llvm/llvm-project.git llvmorg-19.1.7-0-gcd708029e0b2)
Target: x86_64-unknown-freebsd15.0
Thread model: posix
InstalledDir: /usr/bin
```

I checked OpenBSD and it is also using Clang as the default compiler.

I tried to build Ladybird on FreeBSD 15 and got stuck with vcpkg libvpx due to linker error with undefined pthread_create. I set the following variables and build was successful afterwards.

```
export CFLAGS="$CFLAGS -pthread"
export CXXFLAGS="$CXXFLAGS -pthread"
export LDFLAGS="$LDFLAGS -pthread"
```

Ran the Release build and it is working fine viewing several web sites including the default bookmarked ones related to Ladybird, even the icon is working fine in the Plasma desktop.  Just got this when running test

```
        Start  85: TestWeakPtr
 85/193 Test  #85: TestWeakPtr ....................................   Passed    0.01 sec
        Start  86: TestLexicalPath
 86/193 Test  #86: TestLexicalPath ................................   Passed    0.01 sec
        Start  87: TestFunctionObjC
 87/193 Test  #87: TestFunctionObjC ...............................***Failed    0.01 sec
ld-elf.so.1: /home/free/projects/ladybird/Build/release/bin/TestFunctionObjC: Undefined symbol "__gnustep_objcxx_personality_v0"


99% tests passed, 1 tests failed out of 87

Total Test time (real) =   3.46 sec

The following tests FAILED:
         87 - TestFunctionObjC (Failed)
Errors while running CTest
```

Ran test-js successfully, and test-web ran, but I had 

```
==========================================================
Pass: 6732, Fail: 59, Skipped: 226, Timeout: 1, Crashed: 2
==========================================================
Fail: Layout/input/grid/image-in-grid.html
Fail: Text/input/wpt-import/html/canvas/element/path-objects/2d.path.arc.nonfinite.html
Fail: Text/input/wpt-import/html/canvas/element/path-objects/2d.path.arc.zeroradius.html
Fail: Text/input/wpt-import/html/canvas/element/drawing-rectangles-to-the-canvas/2d.clearRect.nonfinite.html
Fail: Text/input/wpt-import/html/canvas/element/compositing/2d.composite.transparent.lighter.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.gradient.empty.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.fillStyle.parse.current.basic.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.fillStyle.parse.current.changed.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.basic.nocontext.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.4.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.transform.identity.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.transform.infinity.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.gradient.transparent.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.3.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.cap.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.5.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.2.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.3.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.enable.off.2.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.2.html
Fail: Text/input/wpt-import/html/canvas/element/manual/drawing-images-to-the-canvas/drawimage_canvas.html
Fail: Text/input/wpt-import/html/canvas/element/manual/fill-and-stroke-styles/2d.fillStyle.parse.current.notrendered.html
Fail: Text/input/wpt-import/html/canvas/element/canvas-host/2d.canvas.host.initial.reset.transform.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.rect2.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-bounds.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.negative.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.zero.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.outside.html
Fail: Text/input/wpt-import/html/canvas/element/canvas-host/2d.canvas.host.initial.reset.clip.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.unaffected.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-drawImage.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.rect1.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.top.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.bottom.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.hanging.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.alphabetic.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.fill.maxWidth.NaN.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.middle.html
Fail: Text/input/wpt-import/dom/nodes/Document-createElementNS.html
Fail: Text/input/wpt-import/css/compositing/canvas-composite-modes.html
Fail: Text/input/wpt-import/css/selectors/attribute-selectors/attribute-case/syntax.html
Fail: Text/input/wpt-import/css/selectors/attribute-selectors/attribute-case/semantics.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-exif-orientation_none.html
Fail: Text/input/wpt-import/html/canvas/offscreen/text/2d.text.font.weight.worker.html
Fail: Text/input/wpt-import/html/canvas/offscreen/text/2d.text.font.relative_size.worker.html
Fail: Text/input/wpt-import/html/semantics/embedded-content/the-img-element/img.complete.html
Fail: Text/input/HTML/CanvasRenderingContext2D-get-image-data-correctness.html
Fail: Text/input/HTML/canvas-getImageData-no-alpha.html
Crashed: Text/input/canvas/export-offscreencanvas.html
Fail: Text/input/canvas/resize-canvas-and-keep-painting.html
Crashed: Text/input/canvas/export.html
Fail: Ref/input/wpt-import/html/the-xhtml-syntax/parsing-xhtml-documents/adopt-while-parsing-001.html
Fail: Screenshot/input/canvas-fillstyle-opacity.html
Fail: Screenshot/input/canvas-fillstyle-rgb.html
Fail: Screenshot/input/canvas-path-rect.html
Fail: Screenshot/input/canvas-fillstyle-gradients.html
Fail: Screenshot/input/canvas-filters.html
Fail: Screenshot/input/canvas-text.html
Fail: Screenshot/input/canvas-shadow.html
Timeout: Text/input/wpt-import/navigation-api/navigation-methods/navigate-same-document.html
Results: file:///home/free/projects/ladybird/test-dumps/results/index.html

```

Ran the Sanitizer and during build has 'angle' build issue with this message

```

-- Found ZLIB: /home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libz.so (found version "1.3.1")
-- Using platform-specific CMakeLists: /home/free/projects/ladybird/Build/vcpkg/buildtrees/angle/src/bd4ca4d539-caeb5ad129.clean/PlatformLinux.cmake
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Error at CMakeLists.txt:371 (WEBKIT_ADD_TARGET_C_FLAGS):
  Unknown CMake command "WEBKIT_ADD_TARGET_C_FLAGS".
```

Exited the terminal to clear out the exports done earlier, and git reset/clean everything to start over fresh for Sanitizer.  Had same libvpx issue again, so reapplied the exports, and then the build was successful.

If I start the Sanitizer build it fails with some other ASAN issue, not related as far as I can tell:

```
free@FreeBSD:~/projects/ladybird $ BUILD_PRESET=Sanitizer ./Meta/ladybird.py run
ninja: Entering directory `/home/free/projects/ladybird/Build/sanitizers'
[0/2] Re-checking globbed directories...
ninja: no work to do.
=================================================================
==76603==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x51d00002074c at pc 0x0000011956d9 bp 0x7fffffffbce0 sp 0x7fffffffb4a0
READ of size 333 at 0x51d00002074c thread T0
    #0 0x0000011956d8 in strndup /usr/src/contrib/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:416:3
    #1 0x00081cdac09f  (/usr/local/lib/libxkbcommon-x11.so.0+0xa09f)
    #2 0x00081cda9b0c  (/usr/local/lib/libxkbcommon-x11.so.0+0x7b0c)
    #3 0x00081cda90dc  (/usr/local/lib/libxkbcommon-x11.so.0+0x70dc)
    #4 0x00081cda8a00 in xkb_x11_keymap_new_from_device (/usr/local/lib/libxkbcommon-x11.so.0+0x6a00)
    #5 0x00081ce8bb0b in QXcbKeyboard::updateKeymap() qxcbkeyboard.cpp
    #6 0x00081ce8cb72 in QXcbKeyboard::QXcbKeyboard(QXcbConnection*) qxcbkeyboard.cpp
    #7 0x00081ce5fe3a in QXcbConnection::QXcbConnection(QXcbNativeInterface*, bool, unsigned int, char const*) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/Qt6/plugins/platforms/../../../lib/libQt6XcbQpa.so.6+0x51e3a)
    #8 0x00081ce88910 in QXcbIntegration::QXcbIntegration(QList<QString> const&, int&, char**) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/Qt6/plugins/platforms/../../../lib/libQt6XcbQpa.so.6+0x7a910)
    #9 0x0008173f87f1 in QXcbIntegrationPlugin::create(QString const&, QList<QString> const&, int&, char**) qxcbmain.cpp
    #10 0x000802d28766 in QPlatformIntegrationFactory::create(QString const&, QList<QString> const&, int&, char**, QString const&) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x328766)
    #11 0x000802cedf3a in init_platform(QString const&, QString const&, QString const&, int&, char**) qguiapplication.cpp
    #12 0x000802ced416 in QGuiApplicationPrivate::createPlatformIntegration() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2ed416)
    #13 0x000802cefe03 in QGuiApplicationPrivate::createEventDispatcher() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2efe03)
    #14 0x000803705fdd in QCoreApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Core.so.6+0x305fdd)
    #15 0x000802ce83fd in QGuiApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2e83fd)
    #16 0x0008016e8061 in QApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Widgets.so.6+0x2e8061)
    #17 0x000001245a44 in LadybirdQApplication /home/free/projects/ladybird/UI/Qt/Application.cpp:63:11
    #18 0x000001245a44 in make<Ladybird::LadybirdQApplication, Main::Arguments &> /home/free/projects/ladybird/AK/NonnullOwnPtr.h:148:59
    #19 0x000001245a44 in Ladybird::Application::create_platform_event_loop() /home/free/projects/ladybird/UI/Qt/Application.cpp:112:25
    #20 0x000801e8ab77 in WebView::Application::initialize(Main::Arguments const&) /home/free/projects/ladybird/Libraries/LibWebView/Application.cpp:369:20
    #21 0x000001244126 in AK::ErrorOr<AK::NonnullOwnPtr<Ladybird::Application>, AK::Error> Ladybird::Application::create<>(Main::Arguments const&) /home/free/projects/ladybird/UI/Qt/Application.h:19:5
    #22 0x000001241db4 in ladybird_main(Main::Arguments) /home/free/projects/ladybird/UI/Qt/main.cpp:46:16
    #23 0x00000134fd33 in main /home/free/projects/ladybird/Libraries/LibMain/Main.cpp:46:19
    #24 0x000810f9537e in __libc_start1 /usr/src/lib/libc/csu/libc_start1.c:180:7
    #25 0x00000117e8c0 in _start /usr/src/lib/csu/amd64/crt1_s.S:80

0x51d00002074c is located 0 bytes after 2252-byte region [0x51d00001fe80,0x51d00002074c)
allocated by thread T2 here:
    #0 0x0000011ef2df in malloc /usr/src/contrib/llvm-project/compiler-rt/lib/asan/asan_malloc_linux.cpp:68:3
    #1 0x00081716a12d  (/usr/local/lib/libxcb.so.1+0x1912d)
    #2 0x0008171682ec  (/usr/local/lib/libxcb.so.1+0x172ec)
    #3 0x00081716a5e8 in xcb_wait_for_event (/usr/local/lib/libxcb.so.1+0x195e8)
    #4 0x00081ce850a3 in QXcbEventQueue::run() qxcbeventqueue.cpp
    #5 0x0008038b69a0 in QThreadPrivate::start(void*) qthread_unix.cpp
    #6 0x0000011ec99a in asan_thread_start(void*) /usr/src/contrib/llvm-project/compiler-rt/lib/asan/asan_interceptors.cpp:239:28
    #7 0x00080e3b4d20 in thread_start /usr/src/lib/libthr/thread/thr_create.c:299:16

Thread T2 created by T0 here:
    #0 0x0000011e20e1 in pthread_create /usr/src/contrib/llvm-project/compiler-rt/lib/asan/asan_interceptors.cpp:250:3
    #1 0x0008038b7768 in QThread::start(QThread::Priority) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Core.so.6+0x4b7768)
    #2 0x00081ce5fd5d in QXcbConnection::QXcbConnection(QXcbNativeInterface*, bool, unsigned int, char const*) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/Qt6/plugins/platforms/../../../lib/libQt6XcbQpa.so.6+0x51d5d)
    #3 0x00081ce88910 in QXcbIntegration::QXcbIntegration(QList<QString> const&, int&, char**) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/Qt6/plugins/platforms/../../../lib/libQt6XcbQpa.so.6+0x7a910)
    #4 0x0008173f87f1 in QXcbIntegrationPlugin::create(QString const&, QList<QString> const&, int&, char**) qxcbmain.cpp
    #5 0x000802d28766 in QPlatformIntegrationFactory::create(QString const&, QList<QString> const&, int&, char**, QString const&) (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x328766)
    #6 0x000802cedf3a in init_platform(QString const&, QString const&, QString const&, int&, char**) qguiapplication.cpp
    #7 0x000802ced416 in QGuiApplicationPrivate::createPlatformIntegration() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2ed416)
    #8 0x000802cefe03 in QGuiApplicationPrivate::createEventDispatcher() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2efe03)
    #9 0x000803705fdd in QCoreApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Core.so.6+0x305fdd)
    #10 0x000802ce83fd in QGuiApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Gui.so.6+0x2e83fd)
    #11 0x0008016e8061 in QApplicationPrivate::init() (/home/free/projects/ladybird/Build/sanitizers/vcpkg_installed/x64-freebsd-dynamic/lib/libQt6Widgets.so.6+0x2e8061)
    #12 0x000001245a44 in LadybirdQApplication /home/free/projects/ladybird/UI/Qt/Application.cpp:63:11
    #13 0x000001245a44 in make<Ladybird::LadybirdQApplication, Main::Arguments &> /home/free/projects/ladybird/AK/NonnullOwnPtr.h:148:59
    #14 0x000001245a44 in Ladybird::Application::create_platform_event_loop() /home/free/projects/ladybird/UI/Qt/Application.cpp:112:25
    #15 0x000801e8ab77 in WebView::Application::initialize(Main::Arguments const&) /home/free/projects/ladybird/Libraries/LibWebView/Application.cpp:369:20
    #16 0x000001244126 in AK::ErrorOr<AK::NonnullOwnPtr<Ladybird::Application>, AK::Error> Ladybird::Application::create<>(Main::Arguments const&) /home/free/projects/ladybird/UI/Qt/Application.h:19:5
    #17 0x000001241db4 in ladybird_main(Main::Arguments) /home/free/projects/ladybird/UI/Qt/main.cpp:46:16
    #18 0x00000134fd33 in main /home/free/projects/ladybird/Libraries/LibMain/Main.cpp:46:19
    #19 0x000810f9537e in __libc_start1 /usr/src/lib/libc/csu/libc_start1.c:180:7
    #20 0x00000117e8c0 in _start /usr/src/lib/csu/amd64/crt1_s.S:80
    #21 0x000801380007  (<unknown module>)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/usr/local/lib/libxkbcommon-x11.so.0+0xa09f) 
Shadow bytes around the buggy address:
  0x51d000020480: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x51d000020500: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x51d000020580: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x51d000020600: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x51d000020680: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x51d000020700: 00 00 00 00 00 00 00 00 00[04]fa fa fa fa fa fa
  0x51d000020780: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x51d000020800: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x51d000020880: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x51d000020900: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x51d000020980: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==76603==ABORTING

```

If I run Sanitizer test I get the same failure as Release with TestFunctionObjC. test-js is OK. test-lib has more fail/timeout/crash than I got in Release.

```
==========================================================
Pass: 6659, Fail: 61, Skipped: 226, Timeout: 62, Crashed: 8
==========================================================
Fail: Layout/input/grid/image-in-grid.html
Fail: Layout/input/hide-audio-based-on-controls.html
Timeout: Text/input/all-window-properties.html
Timeout: Text/input/wpt-import/trusted-types/block-string-assignment-to-DOMParser-parseFromString.html
Timeout: Text/input/wpt-import/trusted-types/TrustedTypePolicyFactory-getPropertyType.tentative.html
Timeout: Text/input/wpt-import/resize-observer/callback-cross-realm-report-exception.html
Timeout: Text/input/wpt-import/html/browsers/origin/api/origin-from-url.any.html
Fail: Text/input/wpt-import/html/canvas/element/drawing-rectangles-to-the-canvas/2d.clearRect.nonfinite.html
Fail: Text/input/wpt-import/html/canvas/element/path-objects/2d.path.arc.zeroradius.html
Fail: Text/input/wpt-import/html/canvas/element/compositing/2d.composite.transparent.lighter.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.gradient.empty.html
Fail: Text/input/wpt-import/html/canvas/element/path-objects/2d.path.arc.nonfinite.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.basic.nocontext.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.fillStyle.parse.current.basic.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.transform.infinity.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.4.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.cap.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.gradient.transparent.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.5.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.fillStyle.parse.current.changed.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.1.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.enable.off.2.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.3.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.1.html
Fail: Text/input/wpt-import/html/canvas/element/manual/fill-and-stroke-styles/2d.fillStyle.parse.current.notrendered.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.2.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.stroke.join.3.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-bounds.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.negative.html
Fail: Text/input/wpt-import/html/canvas/element/shadows/2d.shadow.alpha.2.html
Fail: Text/input/wpt-import/html/canvas/element/fill-and-stroke-styles/2d.pattern.transform.identity.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.unaffected.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.rect2.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.outside.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.rect1.html
Fail: Text/input/wpt-import/html/canvas/element/pixel-manipulation/2d.imageData.put.dirty.zero.html
Fail: Text/input/wpt-import/html/canvas/element/manual/drawing-images-to-the-canvas/drawimage_canvas.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.top.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.middle.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.bottom.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.hanging.html
Fail: Text/input/wpt-import/html/canvas/element/canvas-host/2d.canvas.host.initial.reset.clip.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.fill.maxWidth.NaN.html
Fail: Text/input/wpt-import/html/canvas/element/text/2d.text.draw.baseline.alphabetic.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-drawImage.html
Fail: Text/input/wpt-import/html/canvas/element/canvas-host/2d.canvas.host.initial.reset.transform.html
Timeout: Text/input/wpt-import/wasm/core/call_indirect.wast.js.html
Timeout: Text/input/wpt-import/html/syntax/parsing/html5lib_plain-text-unsafe-write_single.html
Timeout: Text/input/wpt-import/html/syntax/parsing/empty-doctype-ids.html
Timeout: Text/input/wpt-import/html/syntax/parsing/html5lib_tests3.html
Timeout: Text/input/wpt-import/html/semantics/forms/the-option-element/option-value.html
Fail: Text/input/wpt-import/html/canvas/element/manual/imagebitmap/createImageBitmap-exif-orientation_none.html
Timeout: Text/input/wpt-import/html/semantics/embedded-content/the-img-element/img.complete.html
Timeout: Text/input/wpt-import/html/dom/access-key-label.html
Timeout: Text/input/wpt-import/html/the-xhtml-syntax/parsing-xhtml-documents/xhtml-mathml-dtd-entity-12.htm
Timeout: Text/input/wpt-import/html/the-xhtml-syntax/parsing-xhtml-documents/xml-stylesheet-blocking.xhtml
Timeout: Text/input/wpt-import/html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-extra.html
Timeout: Text/input/wpt-import/html/webappapis/dynamic-markup-insertion/document-write/script_011.html
Fail: Text/input/wpt-import/dom/nodes/Document-createElementNS.html
Timeout: Text/input/wpt-import/dom/nodes/Comment-in-doctype.xhtml
Timeout: Text/input/wpt-import/dom/events/Event-stopPropagation-cancel-bubbling.html
Timeout: Text/input/wpt-import/scroll-animations/css/timeline-scope-parsing.html
Fail: Text/input/wpt-import/css/compositing/canvas-composite-modes.html
Timeout: Text/input/wpt-import/webstorage/defineProperty.window.html
Timeout: Text/input/wpt-import/css/css-align/gaps/row-gap-parsing-001.html
Timeout: Text/input/wpt-import/css/css-nesting/nested-declarations-matching.html
Timeout: Text/input/wpt-import/css/css-flexbox/flex-item-contains-strict.html
Timeout: Text/input/wpt-import/css/css-flexbox/alignment/flex-align-baseline-multicol-001.html
Timeout: Text/input/wpt-import/css/css-animations/animation-composition.html
Timeout: Text/input/wpt-import/css/css-typed-om/the-stylepropertymap/properties/transform.html
Timeout: Text/input/wpt-import/css/css-grid/animation/grid-template-rows-interpolation.html
Timeout: Text/input/wpt-import/css/css-backgrounds/animations/box-shadow-interpolation.html
Fail: Text/input/wpt-import/css/css-values/calc-in-media-queries-with-mixed-units.html
Timeout: Text/input/wpt-import/css/css-env/indexed-env.tentative.html
Fail: Text/input/wpt-import/css/selectors/attribute-selectors/attribute-case/syntax.html
Fail: Text/input/wpt-import/css/selectors/attribute-selectors/attribute-case/semantics.html
Timeout: Text/input/wpt-import/css/css-shapes/parsing/shape-margin-valid.html
Timeout: Text/input/wpt-import/css/css-multicol/parsing/columns-invalid.html
Timeout: Text/input/wpt-import/custom-elements/CustomElementRegistry.html
Timeout: Text/input/wpt-import/IndexedDB/delete-request-queue.any.html
Fail: Text/input/wpt-import/xhr/formdata/set-blob.any.html
Fail: Text/input/wpt-import/user-timing/mark.any.html
Timeout: Text/input/wpt-import/pointerevents/parsing/touch-action-computed.html
Fail: Text/input/HTML/CanvasRenderingContext2D-get-image-data-correctness.html
Fail: Text/input/HTML/canvas-getImageData-no-alpha.html
Timeout: Text/input/input-blur.html
Timeout: Text/input/HTML/Window-iterate-properties.html
Timeout: Text/input/HTML/session-storage-event-fired-to-lazy-window.html
Timeout: Text/input/HTML/HTMLObjectElement-form.html
Timeout: Text/input/html/csp-frame-ancestors-opaque-origin.html
Crashed: Text/input/canvas/export-offscreencanvas.html
Fail: Text/input/canvas/resize-canvas-and-keep-painting.html
Crashed: Text/input/canvas/export.html
Timeout: Text/input/indexeddb-result-after-abort-in-upgradeneeded.html
Timeout: Text/input/indexeddb-abort-upgrade-reverts-handle-names.html
Timeout: Text/input/indexeddb-abort-in-success-with-queued-requests.html
Timeout: Text/input/scroll-with-pinch-zoom-applied.html
Timeout: Text/input/canvas/2d-canvas-create-imagedata.html
Timeout: Text/input/createBitmap.html
Timeout: Text/input/Streams/ReadableStreamBYOBReader-read-min.html
Timeout: Text/input/Worker/Worker-script-fetch-referrer.html
Timeout: Text/input/css/shadow-root-slot-ancestor-style-sheet-mutations.html
Timeout: Text/input/indexeddb-transaction-scheduling.html
Fail: Ref/input/wpt-import/html/the-xhtml-syntax/parsing-xhtml-documents/adopt-while-parsing-001.html
Fail: Ref/input/css/images-load-relative-to-imported-style-sheet.html
Timeout: Crash/HTML/multiple-images-iframe-removed.html
Timeout: Crash/HTML/range-input-plain-event.html
Timeout: Crash/HTML/image-error-after-iframe-removed.html
Timeout: Crash/HTML/range-input-step-any-keydown.html
Timeout: Crash/HTML/validity-attribute.html
Timeout: Crash/HTML/invalid-integrity-attribute.html
Timeout: Crash/HTML/range-input-step-any-wheel.html
Timeout: Crash/HTML/media-srcObject-blob.html
Timeout: Crash/HTML/media-load-during-track-setup.html
Timeout: Crash/HTML/image-load-after-iframe-navigated.html
Timeout: Crash/HTML/viewport-client-gc-after-adopt-node.html
Timeout: Crash/HTML/rapid-iframe-image-loading.html
Fail: Screenshot/input/canvas-fillstyle-opacity.html
Fail: Screenshot/input/canvas-path-rect.html
Fail: Screenshot/input/canvas-fillstyle-rgb.html
Fail: Screenshot/input/canvas-fillstyle-gradients.html
Fail: Screenshot/input/canvas-text.html
Fail: Screenshot/input/canvas-filters.html
Timeout: Screenshot/input/acid2.html
Fail: Screenshot/input/canvas-shadow.html
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_tests2.html?run_type=uri
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_namespace-sensitivity.html?run_type=write_single
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_tests12.html?run_type=write_single
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_tables01.html?run_type=write_single
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_tests12.html?run_type=write
Crashed: Text/input/wpt-import/html/syntax/parsing/html5lib_tests6.html?run_type=uri
Results: file:///home/free/projects/ladybird/test-dumps/results/index.html
```

--- a/Libraries/LibJS/AsmIntGen/src/codegen_x86_64.rs
+++ b/Libraries/LibJS/AsmIntGen/src/codegen_x86_64.rs
@@ -561,14 +561,36 @@ fn emit_instruction(
 
         // call_raw_native pseudo-instruction
         // Indirectly calls ThrowCompletionOr<Value> (*)(VM&) from a register.
-        // Passes the hidden VM* as the sole argument. Result payload lands in
-        // rax (t0) and the Variant index is normalized into rcx (t1).
+        // Result payload lands in rax (t0) and the Variant index is normalized
+        // into rcx (t1).
+        //
+        // x86_64 ABI difference when returning ThrowCompletionOr<Value>:
+        // - SysV (ELF: Linux/BSD): returned in rax/rdx.
+        // - Darwin (Mach-O): returned via sret (hidden pointer in rdi).
+        //
+        // Codegen is split based on the object format to match the ABI.
         "call_raw_native" => {
             if let Some(op) = insn.operands.first() {
                 let func = resolve_op(op, handler, program);
                 w!(out, "    mov r11, {func}");
-                w!(out, "    mov rdi, QWORD PTR [rbp - 48]");
-                w!(out, "    call r11");
+                if matches!(program.object_format, ObjectFormat::MachO) {
+                    // sret: hidden pointer in rdi, VM& in rsi. Reserve a 16-byte
+                    // return slot on the stack; rsp is already 16-byte aligned
+                    // at this point, so sub rsp, 16 preserves the alignment the
+                    // call requires.
+                    w!(out, "    sub rsp, 16");
+                    w!(out, "    mov rdi, rsp");
+                    w!(out, "    mov rsi, QWORD PTR [rbp - 48]");
+                    w!(out, "    call r11");
+                    w!(out, "    mov rax, QWORD PTR [rsp]");
+                    w!(out, "    mov rdx, QWORD PTR [rsp + 8]");
+                    w!(out, "    add rsp, 16");
+                } else {
+                    // Register return: VM& in rdi, result lands in rax (payload)
+                    // and rdx (tag).
+                    w!(out, "    mov rdi, QWORD PTR [rbp - 48]");
+                    w!(out, "    call r11");
+                }
                 w!(out, "    mov rcx, rdx");
             }
         }

